src/components/ha-sidebar-panels.ts

```diff
@@ -0,0 +1,344 @@
+    display: flex;
+    flex-direction: column;
+    flex-grow: 1;
+    margin: 0 -12px;
+    padding: 0 12px;
+  }
+  .spacer {
+    flex: 1;
+  }
+`;
+
+const SHOW_AFTER_SPACER = ["config", "developer-tools"];
+const SORT_VALUE_URL_PATHS = {
+  energy: 1,
+  map: 2,
+  logbook: 3,
+  history: 4,
+  "developer-tools": 9,
+  config: 11,
+};
+const PANEL_ICONS = {
+  calendar: mdiCalendar,
+  "developer-tools": mdiHammer,
+  energy: mdiLightningBolt,
+  history: mdiChartBox,
+  logbook: mdiFormatListBulletedType,
+  lovelace: mdiViewDashboard,
+  map: mdiTooltipAccount,
+  "media-browser": mdiPlayBoxMultiple,
+  "shopping-list": mdiCart,
+};
+const panelSorter = (
+  a: PanelInfo,
+  b: PanelInfo,
+  panelOrder: string[],
+  defaultPanel: string,
+  language: string
+) => {
+  const indexA = panelOrder.indexOf(a.url_path);
+  const indexB = panelOrder.indexOf(b.url_path);
+  const order = indexA - indexB;
+  if (order) return order;
+  return defaultPanelSorter(defaultPanel, a, b, language);
+};
+const defaultPanelSorter = (
+  defaultPanel: string,
+  a: PanelInfo,
+  b: PanelInfo,
+  language: string
+) => {
+  if (a.url_path === defaultPanel) return -1;
+  if (b.url_path === defaultPanel) return 1;
+
+  const aLovelace = a.component_name === "lovelace";
+  const bLovelace = b.component_name === "lovelace";
+
+  if (aLovelace && bLovelace)
+    return stringCompare(a.title!, b.title!, language);
+  if (aLovelace) return -1;
+  if (bLovelace) return 1;
+
+  const aBuiltIn = a.url_path in SORT_VALUE_URL_PATHS;
+  const bBuiltIn = b.url_path in SORT_VALUE_URL_PATHS;
+
+  if (aBuiltIn && bBuiltIn) {
+    return SORT_VALUE_URL_PATHS[a.url_path] - SORT_VALUE_URL_PATHS[b.url_path];
+  }
+  if (aBuiltIn) return -1;
+  if (bBuiltIn) return 1;
+
+  return stringCompare(a.title!, b.title!, language);
+};
+const _computePanels = (
+  panels: HomeAssistant["panels"],
+  defaultPanel: HomeAssistant["defaultPanel"],
+  panelsOrder: string[],
+  hiddenPanels: string[],
+  locale: HomeAssistant["locale"]
+): [PanelInfo[], PanelInfo[]] => {
+  if (!panels) {
+    return [[], []];
+  }
+
+  const beforeSpacer: PanelInfo[] = [];
+  const afterSpacer: PanelInfo[] = [];
+
+  Object.values(panels).forEach((panel) => {
+    if (
+      hiddenPanels.includes(panel.url_path) ||
+      (!panel.title && panel.url_path !== defaultPanel)
+    ) {
+      return;
+    }
+    (SHOW_AFTER_SPACER.includes(panel.url_path)
+      ? afterSpacer
+      : beforeSpacer
+    ).push(panel);
+  });
+
+  beforeSpacer.sort((a, b) =>
+    panelSorter(a, b, panelsOrder, defaultPanel, locale.language)
+  );
+  afterSpacer.sort((a, b) =>
+    panelSorter(a, b, panelsOrder, defaultPanel, locale.language)
+  );
+  return [beforeSpacer, afterSpacer];
+};
+const computePanels = memoizeOne(_computePanels);
+
+@customElement("ha-sidebar-panels")
+class HaSidebarPanels extends LitElement {
+  static shadowRootOptions = {
+    ...LitElement.shadowRootOptions,
+    delegatesFocus: true,
+  };
```

Comment:
This is the reason the ordering of dashboards no longer work.

Reply:
Can't reproduce the issue with reordering dashboards
Can you send a screen recording and other relevant info

Reply:
So, I can reproduce this on both Microsoft Edge and Google Chrome latest version. On Mozilla Firefox it works good. I'll see if I can create a screen recording.

Reply:
Can't reproduce w/ Chromium 117.0.5938.132 or Chrome Mobile 118.0.5993.80
Does it work if you clear your localstorage by chance

Reply:
Too bad, removing localstorage doesn't solve it. It works if the shadowRootOptions are removed or delegatesFocus is set to false. Let's wait until somebody other reviews.